### ISSUE TITLE
Error "Model diverged with loss = NaN"

### ISSUE THREAD

**Issue by MoExplorer** [trigeorgis/mdm#12]
To train the model, I have done as README on the github shows several times. But in all training processes, there are two cases that really confuse me:

In the first case, the total training loss is close to zero when the training step comes to 16,000, and then the training process stops because of error "Model diverged with loss = NaN". 

In the second case, the total loss keeps Uunchanged after about 2,000 steps (Does it mean that the model has converged after 2,000 steps?), but the max step is set as 100,000. When the steps come to about 30,000, the training process also stops because of the same error: "Model diverged with loss = NaN"...

I utilize the trained models in both cases to evaluate on the 300W, but the results are a little terrible...For example, the RMSE on lfpw testset is about 18.42%...
 
The only difference from original code is that the extract_patches.cc is compiled into a op library as extract_patches.so, and it is loaded to extract images patches in the training process.

So, could you please explain the two cases to me and tell me how to train the mdm model exactly? I will really appreciate it!
